os/hal/ports/AT32/LLD/DMAv1/at32_dma.h

```diff
@@ -0,0 +1,498 @@
+ */
+#define AT32_DMA_STREAM(id)        (&_at32_dma_streams[id])
+
+#if AT32_DMA1_NUM_CHANNELS > 0
+#define AT32_DMA1_STREAM1          AT32_DMA_STREAM(0)
+#endif
+#if AT32_DMA1_NUM_CHANNELS > 1
+#define AT32_DMA1_STREAM2          AT32_DMA_STREAM(1)
+#endif
+#if AT32_DMA1_NUM_CHANNELS > 2
+#define AT32_DMA1_STREAM3          AT32_DMA_STREAM(2)
+#endif
+#if AT32_DMA1_NUM_CHANNELS > 3
+#define AT32_DMA1_STREAM4          AT32_DMA_STREAM(3)
+#endif
+#if AT32_DMA1_NUM_CHANNELS > 4
+#define AT32_DMA1_STREAM5          AT32_DMA_STREAM(4)
+#endif
+#if AT32_DMA1_NUM_CHANNELS > 5
+#define AT32_DMA1_STREAM6          AT32_DMA_STREAM(5)
+#endif
+#if AT32_DMA1_NUM_CHANNELS > 6
+#define AT32_DMA1_STREAM7          AT32_DMA_STREAM(6)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 0
+#define AT32_DMA2_STREAM1          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 0)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 1
+#define AT32_DMA2_STREAM2          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 1)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 2
+#define AT32_DMA2_STREAM3          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 2)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 3
+#define AT32_DMA2_STREAM4          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 3)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 4
+#define AT32_DMA2_STREAM5          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 4)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 5
+#define AT32_DMA2_STREAM6          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 5)
+#endif
+#if AT32_DMA2_NUM_CHANNELS > 6
+#define AT32_DMA2_STREAM7          AT32_DMA_STREAM(AT32_DMA1_NUM_CHANNELS + 6)
+#endif
+/** @} */
+
+/**
+ * @name    CCTRL register constants common to all DMA types
+ * @{
+ */
+#define AT32_DMA_CCTRL_RESET_VALUE  0x00000000U
+#define AT32_DMA_CCTRL_CHEN         DMA_CCTRL_CHEN
+#define AT32_DMA_CCTRL_DTERRIEN     DMA_CCTRL_DTERRIEN
+#define AT32_DMA_CCTRL_HDTIEN       DMA_CCTRL_HDTIEN
+#define AT32_DMA_CCTRL_FDTIEN       DMA_CCTRL_FDTIEN
+#define AT32_DMA_CCTRL_DTD_MASK     (DMA_CCTRL_DTD | DMA_CCTRL_M2M)
+#define AT32_DMA_CCTRL_DTD_P2M      0U
+#define AT32_DMA_CCTRL_DTD_M2P      DMA_CCTRL_DTD
+#define AT32_DMA_CCTRL_DTD_M2M      DMA_CCTRL_M2M
+#define AT32_DMA_CCTRL_LM           DMA_CCTRL_LM
+#define AT32_DMA_CCTRL_PINCM        DMA_CCTRL_PINCM
+#define AT32_DMA_CCTRL_MINCM        DMA_CCTRL_MINCM
+#define AT32_DMA_CCTRL_PWIDTH_MASK  DMA_CCTRL_PWIDTH
+#define AT32_DMA_CCTRL_PWIDTH_BYTE  0U
+#define AT32_DMA_CCTRL_PWIDTH_HWORD DMA_CCTRL_PWIDTH_0
+#define AT32_DMA_CCTRL_PWIDTH_WORD  DMA_CCTRL_PWIDTH_1
+#define AT32_DMA_CCTRL_MWIDTH_MASK  DMA_CCTRL_MWIDTH
+#define AT32_DMA_CCTRL_MWIDTH_BYTE  0U
+#define AT32_DMA_CCTRL_MWIDTH_HWORD DMA_CCTRL_MWIDTH_0
+#define AT32_DMA_CCTRL_MWIDTH_WORD  DMA_CCTRL_MWIDTH_1
+#define AT32_DMA_CCTRL_SIZE_MASK    (AT32_DMA_CCTRL_PWIDTH_MASK |           \
+                                     AT32_DMA_CCTRL_MWIDTH_MASK)
+#define AT32_DMA_CCTRL_CHPL_MASK    DMA_CCTRL_CHPL
+#define AT32_DMA_CCTRL_CHPL(n)      ((n) << 12U)
+/** @} */
+
```

Comment:
Forgot about this, it needed for I2C and other peripherals.
```suggestion

/**
 * @name    Status flags passed to the ISR callbacks
 * @{
 */
#define AT32_DMA_STS_DTERRF    (0x1U << 3)
#define AT32_DMA_STS_HDTF        (0x1U << 2)
#define AT32_DMA_STS_FDTF         (0x1U << 1)
/** @} */

```